### PR TITLE
Add support to run normal VM

### PR DESCRIPTION
**Pull request by hector-cao**
Add normal VM libvirt template file
Extract domain prefix from the libvirt config file name

### CODE REVIEW

**Review by frenchwr — COMMENTED** [review]
* Based on feedback in https://github.com/canonical/tdx/pull/107, I think we should remove `guest` from the domain naming since it's redundant when talking about a virsh domain or trust domain. My suggestion would be to rename the two template files `trust_domain.xml.template` and `standard_domain.xml.template` and have that reflected in the virsh domain naming convention, for example `trust_domain-abc123...` and `normal_domain-def456...`. What do you think? If we update the naming we should also update the example [in this section of the README](https://github.com/canonical/tdx?tab=readme-ov-file#62-boot-td-with-virsh-libvirt). 
* From the diff of the two templates below, is this (the memoryBacking and launchSecurity configuration) Intel's supported/suggested way to enable a TD? Is that documented some place? 

```diff
$ diff normal_guest.xml.template td_guest.xml.template 
3a4,7
>   <memoryBacking>
>     <source type="anonymous"/>
>     <access mode="private"/>
>   </memoryBacking>
51a56,58
>   <launchSecurity type='tdx'>
>     <policy>0x10000000</policy>
>   </launchSecurity>
```

**Comment by hector-cao** [comment]
> * Based on feedback in [Adjust documentation of Ubuntu 24.04 version #107](https://github.com/canonical/tdx/pull/107), I think we should remove `guest` from the domain naming since it's redundant when talking about a virsh domain or trust domain. My suggestion would be to rename the two template files `trust_domain.xml.template` and `standard_domain.xml.template` and have that reflected in the virsh domain naming convention, for example `trust_domain-abc123...` and `normal_domain-def456...`. What do you think? If we update the naming we should also update the example [in this section of the README](https://github.com/canonical/tdx?tab=readme-ov-file#62-boot-td-with-virsh-libvirt).

Agree

> * From the diff of the two templates below, is this (the memoryBacking and launchSecurity configuration) Intel's supported/suggested way to enable a TD? Is that documented some place?
> 
> ```diff
> $ diff normal_guest.xml.template td_guest.xml.template 
> 3a4,7
> >   <memoryBacking>
> >     <source type="anonymous"/>
> >     <access mode="private"/>
> >   </memoryBacking>
> 51a56,58
> >   <launchSecurity type='tdx'>
> >     <policy>0x10000000</policy>
> >   </launchSecurity>
> ```

Some bits can be found here : https://git.launchpad.net/~kobuk-team/ubuntu/+source/libvirt/tree/debian/patches/tdx/0004-conf-add-tdx-as-launch-security-type.patch?h=tdx&id=5394460408c27941e78a3d41a2d709da4d5e28cb

**Review by frenchwr — COMMENTED** [review]
I vote we remove the `tdvirsh` prefix from the domain names and just use the name from the corresponding template files instead (`trust_domain-123abc` instead of `tdvirsh-trust_domain-123abc`). If you prefer to keep `tdvirsh` perhaps use an underscore instead of a hyphen so it's `tdvirsh_trust_domain...`.

The `./tdvirsh list` example output in the README needs to be updating with the new naming.

I also noticed the output from `./tdvirsh new` shows:

```
Create and run new TD ...
```

even when running the normal template. That could be a bit misleading. How about the following instead? "

```
Create and run new virsh domain from ${TEMPLATE_FILE}
```

Last thing, I promise. :) The command usage states:

```
Run TD VMs managed by libvirt.
```

I think VMs should be removed like we have in other places because it's redundant. Maybe the following instead?

```
Run trust domains managed by libvirt.
```

**Comment by hector-cao** [comment]
> I vote we remove the `tdvirsh` prefix from the domain names and just use the name from the corresponding template files instead (`trust_domain-123abc` instead of `tdvirsh-trust_domain-123abc`). If you prefer to keep `tdvirsh` perhaps use an underscore instead of a hyphen so it's `tdvirsh_trust_domain...`.
> 
> The `./tdvirsh list` example output in the README needs to be updating with the new naming.
> 
> I also noticed the output from `./tdvirsh new` shows:
> 
> ```
> Create and run new TD ...
> ```
> 
> even when running the normal template. That could be a bit misleading. How about the following instead? "
> 
> ```
> Create and run new virsh domain from ${TEMPLATE_FILE}
> ```
> 
> Last thing, I promise. :) The command usage states:
> 
> ```
> Run TD VMs managed by libvirt.
> ```
> 
> I think VMs should be removed like we have in other places because it's redundant. Maybe the following instead?
> 
> ```
> Run trust domains managed by libvirt.
> ```

Ok for all the points except removing the tdvirsh prefix, i think we need it for the `list` and `delete all`operations

**Comment by frenchwr** [comment]
> I would propose the names trust_domain.xml.template and regular_vm.xml.template, or trust_domain.xml.template and vm.xml.template. As mention in PR https://github.com/canonical/tdx/pull/107, "Trust Domain" is the term we coined for a VM protected by Intel TDX. Before, there where just VMs or "regular VMs" to have a more explicit contrast. This is also the language to which I changed the README.
I'm aware that virsh is using the terms "domains", and "guest domains" for every VM managed by it. But at least for me, mixing "Trust Domain" with virsh "guest domain" brings in confusion, because "Trust Domains" and a "regular VMs" are "guest domains" for virsh. "standard domain" is also not used anywhere else.

Thanks, this is helpful. My vote would be for `trust_domain.xml.template` and `regular_vm.xml.template` as it's more explicit, i.e. less chance for confusion.

**Comment by hector-cao** [comment]
> every VM managed by it. But at least for me, mixing "Trust Domain" with virsh "guest domain" brings in confusion, because "Trust Domains" and a "regular VMs" are "guest dom

Ok, fixed.

**Comment by hector-cao** [comment]
> > * Based on feedback in [Adjust documentation of Ubuntu 24.04 version #107](https://github.com/canonical/tdx/pull/107), I think we should remove `guest` from the domain naming since it's redundant when talking about a virsh domain or trust domain. My suggestion would be to rename the two template files `trust_domain.xml.template` and `standard_domain.xml.template` and have that reflected in the virsh domain naming convention, for example `trust_domain-abc123...` and `normal_domain-def456...`. What do you think? If we update the naming we should also update the example [in this section of the README](https://github.com/canonical/tdx?tab=readme-ov-file#62-boot-td-with-virsh-libvirt).
> 
> I would propose the names `trust_domain.xml.template` and `regular_vm.xml.template`, or `trust_domain.xml.template` and `vm.xml.template`. As mention in PR #107, "Trust Domain" is the term we coined for a VM protected by Intel TDX. Before, there where just VMs or "regular VMs" to have a more explicit contrast. This is also the language to which I changed the README.
> 
> I'm aware that virsh is using the terms "domains", and "guest domains" for every VM managed by it. But at least for me, mixing "Trust Domain" with virsh "guest domain" brings in confusion, because "Trust Domains" and a "regular VMs" are "guest domains" for virsh. "standard domain" is also not used anywhere else.

Thanks @BFuhry , agree

File: guest-tools/tdvirsh [hunk 1]
```diff
@@ -24,7 +31,7 @@ usage() {
     cat <<EOM
 Usage: ./$(basename "${BASH_SOURCE[0]}") [-h] [new] [delete D] [list]
 
-Run TD VMs managed by libvirt.
+Run trust domains (Intel TDX) managed by libvirt.
```
[the remark this answers is not in the document]
Comment (hector-cao) [reply guest-tools/tdvirsh:34]: Ok, agree

File: guest-tools/tdvirsh [hunk 2]
```diff
@@ -24,7 +31,7 @@ usage() {
...
 
 Available options:
```
[the remark this answers is not in the document]
Comment (hector-cao) [reply guest-tools/tdvirsh:37]: Agree

File: guest-tools/regular_domain.xml.template [hunk 3]
```diff
@@ -0,0 +1,60 @@
+<domain type='kvm' xmlns:qemu='http://libvirt.org/schemas/domain/qemu/1.0'>
```
Comment (frenchwr) [inline guest-tools/regular_domain.xml.template:1]: I think we agreed to rename this file `regular_vm.xml.template`.

**Review by frenchwr — COMMENTED** [review]
Still need to update the sample output in the README for `tdvirsh --list all` [here](https://github.com/canonical/tdx?tab=readme-ov-file#62-boot-td-with-virsh-libvirt) to match the new naming convention.

**Comment by hector-cao** [comment]
> Still need to update the sample output in the README for `tdvirsh --list all` [here](https://github.com/canonical/tdx?tab=readme-ov-file#62-boot-td-with-virsh-libvirt) to match the new naming convention.

I can also do for this one but basically this command will list all the VMs (regular and TDs)

**Comment by hector-cao** [comment]
> Still need to update the sample output in the README for `tdvirsh --list all` [here](https://github.com/canonical/tdx?tab=readme-ov-file#62-boot-td-with-virsh-libvirt) to match the new naming convention.

I can also do for this one but basically this command will list all the VMs (regular and TDs)

File: guest-tools/regular_domain.xml.template (hunk 3 above)
Comment (hector-cao) [reply guest-tools/regular_domain.xml.template:1]: Ok, missed that

**Comment by frenchwr** [comment]
> > Still need to update the sample output in the README for `tdvirsh --list all` [here](https://github.com/canonical/tdx?tab=readme-ov-file#62-boot-td-with-virsh-libvirt) to match the new naming convention.
> 
> I can also do for this one but basically this command will list all the VMs (regular and TDs), what do you think ?

I think it's fine and intuitive for it to show any TDs or regular VMs created by tdvirsh.

**Review by frenchwr — COMMENTED** [review]
Looks good except we eventually landed on `regular_vm.xml.template` rather than `regular_domain.xml.template`.

**Comment by hector-cao** [comment]
> Looks good except we eventually landed on `regular_vm.xml.template` rather than `regular_domain.xml.template`.

Fixed, thanks. I was lost among all the comments

**Review by frenchwr — APPROVED** [review]
:partying_face:

**Pull request merged**